python-packages/order_utils/src/zero_ex/abi_utils.py

```diff
@@ -0,0 +1,113 @@
+"""Ethereum ABI utilities.
+
+Builds on the eth-abi package, adding some convenience methods like those found
+in npmjs.com/package/ethereumjs-abi.  Ideally, all of this code should be
+pushed upstream into eth-abi.
+"""
+
+import re
+from typing import Any, List
+
+from mypy_extensions import TypedDict
+
+from eth_abi import encode_abi  # type: ignore
+from web3 import Web3  # type: ignore
+
+
+class MethodSignature(TypedDict, total=False):
+    """Object interface to an ABI method signature."""
+
+    method: str
+    args: List[str]
+
+
+def parse_signature(signature: str) -> MethodSignature:
+    """Parse a method signature into its constituent parts.
+
+    >>> parse_signature("ERC20Token(address)")
+    {'method': 'ERC20Token', 'args': ['address']}
+    """
+    if not isinstance(signature, str):
+        raise TypeError(
+            "signature should be str, not " + type(signature).__name__
+        )
```

Comment:
Can we move this to an `assert` module? And just call `assert.isString('signature', signature)` here? We will want to do this exact same thing elsewhere. This way we also don't copy-paste this boilerplate error message in 100 places.

Reply:
[Done](https://github.com/0xProject/0x-monorepo/pull/1144/commits/08e8da00467b1bfa88d9ccbb905aaa31fc9e3b8c)